src/node.js

```diff
@@ -388,6 +393,58 @@
       nextTickQueue.push(obj);
       tickInfo[kLength]++;
     }
+
+    function emitPendingUnhandledRejections() {
+      var hadListeners = false;
+      while (pendingUnhandledRejections.length > 0) {
+        var promise = pendingUnhandledRejections.shift();
+        var reason = pendingUnhandledRejections.shift();
+        if (hasBeenNotifiedProperty.get(promise) === false) {
+          hasBeenNotifiedProperty.set(promise, true);
+          if (!process.emit('unhandledRejection', reason, promise)) {
```

Comment:
No, it is not. Choosing to deal with unhandled rejections like uncaught exceptions is a very opinionated choice.

I'm not saying I agree or disagree with it but it is a breaking change for many users regardless - and not one we should easily introduce. The current plan is to introduce these events now and then choose a default action later - one such default action is to throw the error for `"uncaughtException"` events to deal with it - there are people who like it and people who object to it. However, at first we need the very basic ability to deal with these events so this pull request makes the minimal changes required that don't make any breaking changes to existing io.js code.